sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationClient_private.cs

```diff
@@ -160,7 +160,7 @@ internal static void BuildBatchQuery(RequestUriBuilder builder, SettingSelector
 
             if (selector.Fields != SettingFields.All)
             {
-                var filter = selector.Fields.ToString().ToLowerInvariant();
+                var filter = selector.Fields.ToLowerCaseString();
```

Comment:
Would it be possible just to take the string obtained from ToString().ToLowerInvariant() and replace any instance of "readonly" with "locked" ?  That reduces the test surface a lot, and clarifies precisely what substitution we're making.

Reply:
`string.Replace` is generally a bug prone. We need to control that no words that contain `readonly` are changed. If, for example, in future `ReadOnly` is renamed into `IsReadOnly`, simple `string.Replace` code won't fail but produce `Islocked`.

IMO, serialized strings should be defined explicitly to avoid any glitches. Good solution should probably be done through attributes applied to enum fields, e.g.:
```cs
    /// <summary>
    /// Fields to retrieve from a configuration setting.
    /// </summary>
    [Flags]
    public enum SettingFields : uint
    {
        /// <summary>
        /// A value indicating whether the configuration setting is readonly.
        /// </summary>
        [StringValue("locked")]
        ReadOnly = 0x0040,
```

But since we're close to escrow, I can make it one-line fix for now and have a separate PR for this fix.

Reply:
If just adding the StringValue property to the SettingFields enum will do the trick, I am fine with that.

Reply:
It won't by itself, we will have to write the infrastructure as well :)

Reply:
FWIW, changing the name of an existing enum field would be a breaking change, so not allowed.  I doubt that we'll be adding another field with "readonly" as a substring, so we should be safe, but yes, let's file an issue to track it.  Thanks!